tools/build_rules/workspace.bzl

```diff
@@ -0,0 +1,78 @@
+load('//tools/build_rules:scala.bzl', 'scala_import')
+
+def _scala_maven_jar_impl(repository_ctx):
+    repository_ctx.file(
+        'BUILD.bazel',
+        content = "load('@//tools/build_rules:scala.bzl', 'scala_import')\nscala_import(name='scala_jar', jar='%s', visibility=['//visibility:public'])" % repository_ctx.attr.jar,
+        executable = False,
+    )
+
+scala_maven_jar = repository_rule(
+    implementation = _scala_maven_jar_impl,
+    attrs = {
+        "jar": attr.label(),
+    },
+    local = True,
+)
+
+def declare_maven(hash):
```

Comment:
this file is confusing to me :)
I'd expect scala_maven_jar to be in its own `scala_maven_jar.bzl` file, and the `declare_maven` to be defined in //:WORKSPACE and be passed into maven_repositories() ?

Reply:
> I'd expect scala_maven_jar to be in its own scala_maven_jar.bzl file
That's a good idea. Fixing.

> declare_maven to be defined in //:WORKSPACE and be passed into maven_repositories() 
Oops. It should be. I broke that when splitting up the commits. Fixing.